### DISCUSSION TITLE
Disabling cuda during inference for transformer-ner model

### DISCUSSION THREAD

**Discussion by spatiebalk** [explosion/spaCy#9925] [Help: Installation] [answered]
I have trained a spacy model with the following components [sentencizer, transformers, ner] in Azure ML Studio using a GPU. When I load and run the model locally I can use the model for inference without any GPU (I don't have one) without any changes. 
However, I want to call the registered version of the model in Azure and then apply it to some data stored on Azure, but still run the code for that locally. When doing that I get the error: 

> `...\lib\site-packages\torch\autocast_mode.py:141: UserWarning: User provided device_type of 'cuda', but CUDA is not available. Disabling warnings.warn('User provided device_type of \'cuda\', but CUDA is not available. Disabling')`

After which my code stops running. I already tried to set spacy.require_cpu() as suggested in discussion [#7622](https://github.com/explosion/spaCy/discussions/7622) before calling spacy.load(model), but this does not affect the error at all. 
I also tried setting the cuda visible devices like said [here](https://stackoverflow.com/questions/53266350/how-to-tell-pytorch-to-not-use-the-gpu), but without any effect.

`os.environ['CUDA_VISIBLE_DEVICES']=""`

I'm using:
python 3.8.2
spacy 3.2.1
spacy-transformers 1.1.3
transformers 4.12.5
torch 1.10.1

**Comment by adrianeboyd** [comment]
This probably just related to how torch was installed. Try uninstalling all packages that start with `torch` and reinstalling based on the instructions from their quickstart with the CPU option: https://pytorch.org/get-started/locally/

**Reply by spatiebalk** [reply]
Thank you for your answer. I created a new virtual env and reinstalled all the packages and using the quickstart cpu option for torch. But unfortunately, this didn't affect the error occurring at all.

**Comment by spatiebalk** [comment, marked as answer]
Following [this post](https://github.com/pytorch/pytorch/issues/67598) I downgraded from torch 1.10.1 to 1.9.0 making the error go away.

**Reply by adrianeboyd** [reply]
Thanks for the posting the solution!
